bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/api/IntesisBoxSocketApi.java

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.intesis.internal.api;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketAddress;
+import java.net.UnknownHostException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.thing.ThingStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class handling the Socket connections.
+ *
+ * @author Cody Cutrer - Initial contribution
+ * @author Hans-Jörg Merk - Moved Socket to it's own class
+ */
+@NonNullByDefault
+public class IntesisBoxSocketApi {
+
+    private final Logger logger = LoggerFactory.getLogger(IntesisBoxSocketApi.class);
+
+    private final String ipAddress;
+    private final int port;
+
+    private @Nullable IntesisSocket tcpSocket = null;
+    private @Nullable OutputStreamWriter tcpOutput = null;
+    private @Nullable BufferedReader tcpInput = null;
+
+    private @Nullable IntesisBoxChangeListener changeListener;
+
+    private boolean connected = false;
+
+    public IntesisBoxSocketApi(final String ipAddress, final int port) {
+        this.ipAddress = ipAddress;
+        this.port = port;
+    }
+
+    private class IntesisSocket {
```

Comment:
Why is this class needed? It doesn't add any functionality so just keep track of the Socket instead.

Reply:
That's how I saw it in the energenie Binding, which socket class was written by @Hilbrand .  I have no clue of socket handling, so it was simply copy and paste.